File: potenza.c
#include <stdio.h>

int potenza(int b, int e);

int main(void){
    
    int b;
    int e;

    printf("Potenza\n\n");

    printf("Inserisci la base: ");
    scanf("%d", &b);

    do {
        printf("Inserisci l'esponente: ");
        scanf("%d", &e);
    }while (e < 0);

    printf("%d^%d = %d", b, e, potenza(b, e));

    putchar('\n');

    return 0;
}

int potenza(int b, int e){

    if (e == 1) {
        return b;
    }
    else if (e == 0) {
        return 1;
    }

    return b * potenza(b, e-1);
}

File: potenza.py
def potenza(b, e):
	if e == 1:
		return b
	elif e == 0:
		return 1

	return b * potenza(b, e-1)

# main
e = -1
while e < 0:
	b, e = input("Inserisci la potenza (es. 3^2): ").split('^')
	b = int(b)
	e = int(e)

power = potenza(b, e)

print(f"{b}^{e} = {power}\n")